File: sort colours.java
//timecomplexity :-o(n);
//space complexity:-0(1);
//did it run on leetcode:- yes
//any problem you faced:- got wrong answer manytimes.
// your code with explanation:- as there are only three colours if the element is 2 swapping to the right most and decrementing, if it is zero swping with previous. 
//simply keeping low as 0 boundaries and high as 2 boundary.






class Solution {
    public void sortColors(int[] nums) {
        int low=0;
        int high=nums.length-1;
        int current=0;
        while(current<=high){
            if(nums[current]==2){
                int temp=nums[current];
                nums[current]=nums[high];
                nums[high]=temp;
                high=high-1;
            }
            else if(nums[current]==0){
                int temp1=nums[current];
                nums[current]=nums[low];
                nums[low]=temp1;
                low=low+1;
                current=current+1;
            }
            else{
                current=current+1;
            }
        }
        
        
        
    }
}

File: SortColours.py
from pip import List
#Time Complexity = O(n)
#Space Complexity = O(1)

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n = len(nums)
        low, mid, high = 0,0,n-1
        while(low<high and mid<=high):
            if(nums[mid]==0):
                nums[low], nums[mid]= nums[mid], nums[low]
                low+=1
                mid+=1
            elif(nums[mid]==1):
                mid+=1
            else:
                nums[mid], nums[high] = nums[high], nums[mid]
                high-=1
        
        
        